### PR TITLE
Basic autogeneration

### PR DESCRIPTION
**Pull request by pvienhage**
- [ ] Tag the PR with `wip` while in development.
- [ ] Assign yourself as to the PR
- [ ] Assign relevant labels such as `bug`, `enhancement`.
- [ ] Request reviews if the PR is large, complex or you would like an extra pair of eyes to go over it.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [ ] Add new entries to the `Changelog.md`.
- [ ] Update version numbers as needed.

Semver: https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md

https://semver.org/

### CODE REVIEW

File: crypto/stark/src/autogen_sol.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    _public: &[&RationalExpression],
+    constraints: &[RationalExpression],
+    _n_rows: usize,
+    n_cols: usize,
+    blowup: usize,
+) {
+    let mut traces = BTreeMap::new();
+
+    for exp in constraints.iter() {
+        traces.extend(exp.trace_search());
```
Comment (Recmo) [inline crypto/stark/src/autogen_sol.rs:19]: Move the `trace_arguments`, `soldity_encode` and the `*_search` functions to this file. You can add a `impl RationalExpression {` block here (Rust allows many `impl` blocks that don't have to be in the same module)

Comment (Recmo) [reply crypto/stark/src/autogen_sol.rs:19]: Unless we plan to deprecate those in favor of DAG later, in which case we can just remove them

File: crypto/stark/src/autogen_sol.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    return {};
+    ",
+        rows[rows.len() / 2],
+        binary_row_search_string(rows.get(0..(rows.len()) / 2).unwrap()),
+        binary_row_search_string(rows.get((rows.len() / 2)..rows.len()).unwrap()),
+        rows[rows.len() / 2]
+    )
+}
+
+fn lexicographic_compare(first: &RationalExpression, second: &RationalExpression) -> Ordering {
```
Comment (Recmo) [inline crypto/stark/src/autogen_sol.rs:179]: Can we instead `derive(Ord)` on `RationalExpression`? Or `impl Ord on RationalExpression` ?

Comment (pvienhage) [reply crypto/stark/src/autogen_sol.rs:496]: It already exists, and that implementation won't work with this one.

**Pull request merged**